src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -148,45 +148,96 @@ private void DoWorkExpectCancel(object sender, DoWorkEventArgs e)
             manualResetEvent3.Set();
         }
 
+        #endregion
+
         [Fact]
         public void TestThrowExceptionInDoWork()
         {
-            var bw = new TestWorker();
-            bw.DoWork += DoWorkWithException;
-            bw.RunWorkerAsync("Exception");
+            const string expectedArgument = "Exception";
+            const string expectedExceptionMsg = "Exception from DoWork";
+            var bw = new BackgroundWorker();
+
+            bw.DoWork += (sender, e) =>
+            {
+                Assert.Same(bw, sender);
+                Assert.Same(expectedArgument, e.Argument);
+                throw new TestException(expectedExceptionMsg);
+            };
+
+            bw.RunWorkerCompleted += (sender, e) =>
+            {
+                TestException ex = Assert.Throws<TestException>(() => e.Result);
+                Assert.Equal(expectedExceptionMsg, ex.Message);
+            };
+
+            bw.RunWorkerAsync(expectedArgument);
+        }
+
+        [Fact]
+        public void CtorTest()
+        {
+            var bw = new BackgroundWorker();
+            Assert.False(bw.IsBusy);
+            Assert.False(bw.WorkerReportsProgress);
+            Assert.False(bw.WorkerSupportsCancellation);
+            Assert.False(bw.CancellationPending);
         }
 
-        private void DoWorkWithException(object sender, DoWorkEventArgs e)
+        [Fact]
+        public void RunWorkerAsyncTwice()
+        {
+            var bw = new BackgroundWorker();
+            bw.DoWork += (sender, e) => Wait(3 * TimeoutShort);
```

Comment:
What's the value of the waiting here?  If the goal of the test is to invoke RunWorkerAsync while it's already running, then this kind of timing issue can lead to random failures (plus it can result in the tests taking longer than is needed), and it'd be better to use synchronization to ensure you get the interleaving you want, e.g.
```
var bw = new BackgroundWorker();
Barrier barrier = new Barrier();
b.DoWork += (sender, e) => { barrier.SignalAndWait(); barrier.SignalAndWait(); };
bw.RunWorkerAsync();
barrier.SignalAndWait();
Assert.Throws<InvalidOperationException>(() => bw.RunWorkerAsync());
barrier.SignalAndWait();
```